src/main/java/com/softserve/academy/spaced/repetition/service/impl/CourseServiceImpl.java

```diff
@@ -234,23 +227,21 @@ public Course addDeckToCourse(Long courseId, Long deckId) {
 
     @Override
     @Transactional
-    public void updateCoursePrice(PriceDTO priceDTO, Long courseId) {
+    public void updateCoursePrice(Integer price, Long courseId) {
         Course course = courseRepository.findOne(courseId);
-        course = checkIfCoursePriceExists(course);
-        course.getCoursePrice().setPrice(priceDTO.getPrice());
-        coursePriceRepository.save(course.getCoursePrice());
-        courseRepository.save(course);
-    }
-
-    @Override
-    public Course checkIfCoursePriceExists(Course course) {
-        CoursePrice coursePrice;
-        if (course.getCoursePrice() == null) {
-            coursePrice = new CoursePrice();
+        if (price == null) {
+            coursePriceRepository.deleteCoursePriceByCourseId(courseId);
+        } else if (course.getCoursePrice() == null) {
+            CoursePrice coursePrice = new CoursePrice();
             coursePrice.setCourse(course);
+            coursePrice.setPrice(price);
             course.setCoursePrice(coursePrice);
             coursePriceRepository.save(coursePrice);
+        } else if (course.getCoursePrice() != null) {
+            CoursePrice coursePrice = coursePriceRepository.findByCourseId(courseId);
+            coursePrice.setPrice(price);
+            coursePriceRepository.save(coursePrice);
```

Comment:
Use the idea from #346 and remove the coursePriceRepository

Reply:
Removed CoursePriceRepository